samples/emoji-search/android/src/main/java/app/cash/zipline/samples/emojisearch/theme.kt

```diff
@@ -54,12 +65,22 @@ private val Shapes = Shapes(
 
 @Composable
 fun EmojiSearchTheme(
+  darkTheme: Boolean = isSystemInDarkTheme(),
   content: @Composable () -> Unit
 ) {
+  val colors = if (darkTheme) {
+    DarkColorPalette
+  } else {
+    LightColorPalette
+  }
   MaterialTheme(
-    colors = DarkColorPalette,
+    colors = colors,
     typography = Typography,
-    shapes = Shapes,
-    content = content
-  )
+    shapes = Shapes
+  ) {
+    CompositionLocalProvider(
+      LocalContentColor provides MaterialTheme.colors.onBackground,
```

Comment:
This is surprising. Where can I learn about why this is needed?

Reply:
This surprised me too. In the [definition of Text()](https://androidx.tech/artifacts/compose.material/material/1.3.0-rc01-source/androidx/compose/material/Text.kt.html), the default value for `color` is `Color.Unspecified`, and:
```
 * @param color [Color] to apply to the text. If [Color.Unspecified], and [style] has no color set,
 * this will be [LocalContentColor].
```

But I've never noticed this problem before. In a Hello World compose app, even in dark mode, the default background for the activity is still white. But when the activity's theme in the manifest is based on `Theme.MaterialComponents.DayNight` (like it is in emoji-search), the activity background is automatically changed to darkgray/white without Compose.

Reply:
The other solution is obviously to specify the color in the `Text()` composable. Do you have plans on how to bake this into redwood?
```
Text(
    text = "Hello world",
    color = MaterialTheme.colors.onBackground
)
```

Reply:
We can add that to this sample's Compose UI Text binding

Reply:
Yep ok. As long as we always want `ComposeUiText` in the sample app to be the `onBackground` color that works too. Updated the PR.